common/persistence/sql/sqlExecutionManager.go

```diff
@@ -1880,23 +1885,92 @@ func createTimerTasks(
 	return nil
 }
 
-func assertAndUpdateCurrentExecution(tx sqldb.Tx, shardID int, domainID sqldb.UUID, workflowID string, newRunID sqldb.UUID, previousRunID sqldb.UUID,
+func assertNotCurrentExecution(tx sqldb.Tx, shardID int, domainID sqldb.UUID, workflowID string, runID sqldb.UUID) error {
+
+	assertFn := func(currentRow *sqldb.CurrentExecutionsRow) error {
+		if bytes.Equal(currentRow.RunID, runID) {
+			return &p.ConditionFailedError{Msg: fmt.Sprintf(
+				"Assertion on current record failed failed. Current run ID was %v, expected %v",
+				currentRow.RunID,
```

Comment:
this log is misleading - expected and got will always be same - I think you want to return error saying that the records are expected to be different but they are same